pkg/reconciler/v1alpha1/taskrun/resources/image_exporter.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"encoding/json"
+	"flag"
+	"os"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/names"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	imageDigestExporterImage = flag.String("imagedigest-exporter-image", "override-with-imagedigest-exporter-image:latest", "The container image containing our image digest exporter binary.")
+)
+
+// AddOutputImageDigestExporter add a step to check the index.json for all output images
+func AddOutputImageDigestExporter(
+	tr *v1alpha1.TaskRun,
+	taskSpec *v1alpha1.TaskSpec,
+	gr GetResource,
+	logger *zap.SugaredLogger,
+) error {
+
+	output := []*v1alpha1.ImageResource{}
+	if len(tr.Spec.Outputs.Resources) > 0 {
+		for _, trb := range tr.Spec.Outputs.Resources {
+			boundResource, err := getBoundResource(trb.Name, tr.Spec.Outputs.Resources)
+			if err != nil {
+				logger.Errorf("Failed to get bound resource: %s", err)
```

Comment:
what do you think about returning an error that explains the context and letting the caller log instead of passing `logger` into this function?

Reply:
This is the same pattern used in the other functions for `input_resources` and `output_resources`, but anyways this func is returning the error after it logs 

I okay to change it if you want

Reply:
I'd prefer to return an informative error but maybe @vdemeester has a different opinion :)

Reply:
I will change it to return a more informative error